Readme.md

```diff
@@ -484,7 +439,41 @@ class OrganizationFinder extends AbstractDoctrineFinder
 }
 ```
 
-... todo: more explanations
+### How to implement you own event listener ?
+
+To implement other custom event listeners you only need to implement the `EventListenerInterface`:
+
+```php
+class ConsoleOutputListener implements EventListenerInterface
+{
+    /**
+     * @var SystemLoggerInterface
+     * @Flow\Inject
+     */
+    protected $systemLogger;
+
+    /**
+     * @param ButtonTagged $event
+     * @param EventMetadata $metadata
+     */
+    public function whenButtonTagged(ButtonTagged $event, RawEvent $rawEventData)
+    {
+        $this->systemLogger->log('--- ConsoleOutputListener say something has been tagged ---');
+    }
+}
+```
+
+The event handler locator can throw an exception if something is wrong with your command handler definition. In that
+case please check your system log for more information.
+
+All the wiring between event and event listeners is done automatically, if you respect the following convention:
+
+* Method name must be when[ShortEventName]
+* The first parameter must be of type `EventInterface` and is your concrete DomainEvent instance
+* The second parameter is optional and is of type `RawEvent`, containing the raw data and metadata from the EventStore
+
+Also, you can optionally let your `EventListener` implement `ActsBeforeInvokingEventListenerMethodsInterface`
+in order to receive a hook method `beforeInvokingEventListenerMethod` before the concrete event handling method (`when*`) is called.
```

Comment:
It's quite exotic, but this method allows us to prepare *some things* before the actual event handler (method) is called. In the process manager I use this for setting `$this->processState` with the corresponding state so that a user's `whenSomething()` method can use it right away without calling a helper method first. You can see an example in `AbstractProcessManager`.

I think that this interface is still helpful, but please be aware that the Process Manager in its current for is not really optimal and needs to change.